ocaml/xapi/xapi_pool.ml

```diff
@@ -2857,3 +2870,61 @@ let sync_updates ~__context ~self ~force =
         ~doc:"pool.sync_updates"
         ~op:`sync_updates
         (fun () -> create_pool_repository ~__context ~self:repository))
+
+let get_updates_handler (req : Http.Request.t) s _ =
+  debug "Pool.get_updates_handler: received request";
+  req.Http.Request.close <- true;
+  let query = req.Http.Request.query in
+  Xapi_http.with_context "Getting available updates" req s (fun __context ->
+      let all_hosts = Db.Host.get_all ~__context in
+      let hs =
+        match List.assoc "host_refs" query with
+          | v ->
+            List.map (fun ref_str -> Ref.of_string ref_str) (Astring.String.cuts ~sep:";" v)
+            |> (fun l ->
+                let is_invalid ref =
+                    if not (Db.is_valid_ref __context ref && List.mem ref all_hosts) then true
+                    else false
+                in
+                match (List.exists is_invalid l), l with
+                | true, _ | false, [] -> None
+                | false, l -> Some l)
+          | exception Not_found -> Some all_hosts
+      in
+      match hs with
+      | Some hosts ->
+        let codes_of_404 =
+          [
+            Api_errors.no_repository_enabled;
+            Api_errors.invalid_repomd_xml;
+            Api_errors.invalid_updateinfo_xml
+          ]
+        in
+        Xapi_pool_helpers.with_pool_operation
+          ~__context
+          ~self:(Helpers.get_pool ~__context)
+          ~doc:"pool.get_updates"
+          ~op:`get_updates
+          (fun () ->
+             try
+               Repository.with_pool_repository
+                 (fun () ->
+                    let json_str = Yojson.Basic.pretty_to_string
```

Comment:
Why are we pretty printing json just before sending it over the wire?

I would have expected this function to be more suitable:

```
val to_string : ?⁠buf:Bi_outbuf.t -> ?⁠len:int -> ?⁠std:bool -> t -> string
(** Write a compact JSON value to a string*)
```